plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetTypeTranslator.java

```diff
@@ -36,20 +40,38 @@ public final class ParquetTypeTranslator
 {
     private ParquetTypeTranslator() {}
 
-    public static Optional<TypeCoercer<? extends Type, ? extends Type>> createCoercer(PrimitiveTypeName fromParquetType, LogicalTypeAnnotation typeAnnotation, Type toTrinoType)
+    public static Optional<TypeCoercer<? extends Type, ? extends Type>> createCoercer(PrimitiveType fromParquetType, Type toTrinoType)
     {
+        PrimitiveTypeName typeName = fromParquetType.getPrimitiveTypeName();
+        LogicalTypeAnnotation logicalTypeAnnotation = fromParquetType.getLogicalTypeAnnotation();
         if (toTrinoType instanceof DoubleType) {
-            if (isIntegerAnnotationAndPrimitive(typeAnnotation, fromParquetType)) {
-                if (fromParquetType == INT32) {
+            if (isIntegerAnnotationAndPrimitive(logicalTypeAnnotation, typeName)) {
+                if (typeName == INT32) {
                     return Optional.of(new IntegerNumberToDoubleCoercer<>(INTEGER));
                 }
-                if (fromParquetType == INT64) {
+                if (typeName == INT64) {
                     return Optional.of(new IntegerNumberToDoubleCoercer<>(BIGINT));
                 }
             }
         }
         if (toTrinoType instanceof VarcharType varcharType) {
-            if (fromParquetType == INT96) {
+            if (isIntegerAnnotationAndPrimitive(logicalTypeAnnotation, typeName)) {
+                if (typeName == INT32) {
+                    if (logicalTypeAnnotation instanceof LogicalTypeAnnotation.IntLogicalTypeAnnotation intLogicalTypeAnnotation && intLogicalTypeAnnotation.isSigned()) {
+                        Type fromType = switch (intLogicalTypeAnnotation.getBitWidth()) {
+                            case 8 -> TINYINT;
+                            case 16 -> SMALLINT;
+                            case 32 -> INTEGER;
+                            default -> throw new UnsupportedOperationException("Unsupported bit width " + intLogicalTypeAnnotation.getBitWidth());
+                        };
+                        return Optional.of(new IntegerNumberToVarcharCoercer<>(fromType, varcharType));
+                    }
```

Comment:
We can just use `return Optional.of(new IntegerNumberToVarcharCoercer<>(INTEGER, varcharType));` directly

Reply:
For all INT_32 types - Does we return `IntArrayBlock` ? Bcoz if we are planning to pass `IntegerNumberToVarcharCoercer` with `INTEGER` - What if the underlying data is of TINYINT ?

Reply:
In parquet INT32 is the minimum primitive type width for numbers. The logical annotations are just a guarantee by the writer about the range of the written data, it's not relevant for us when we want to read it as a string.

Reply:
So if we have a parquet column of tinyint type (specified in Hive table schema) - Does parquet returns the block as column as `IntArrayBlock` ?

This is bcoz the `IntegerNumberToVarcharCoercer` logic depends on `fromType#getLong` and then converts it to Slice. If we pass `INTEGER` for all INT_32 types - won't it fail when we try to invoke `IntegerType#getLong` as it requires the underlying block to be of `IntArrayBlock`

Reply:
For tinyint, smallint and integer - we return `Optional.of(new IntegerNumberToVarcharCoercer<>(INTEGER, varcharType));`. I'm a bit worried about memory consumption as for tinyint/smallint we tend to consume more memory

Reply:
Parquet stores tiny/small/int as INT32, so we always have to read them as ints which are then copied into bye/short/intarrayblock (with range check if needed). Here we will directly populate INT32 into IntArrayBlock (regardless of logical annotation) and then convert that into varchar. The extra memory consumption is temporary as we will immediately convert to varchar block (which will probably use more memory anyway) and it saves CPU by  reading ints directly instead of copying them into narrower type with range checks.